### PR TITLE
intel_adsp: adsp_memory: add lpsram and hpsram struct definition

### PR DESCRIPTION
**Pull request by dnikodem**
Added new structures and definitions for power management control and status in adsp_memory.h.

This change will allow the removal of address definitions for lpsram and hpsram outside of the Zephyr part.

### CODE REVIEW

File: soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h [hunk 1]
```diff
@@ -65,5 +65,39 @@
...
+	uint32_t USxRMCTL;
+	/** @brief power gating status */
+	uint32_t USxPGISTS;
+};
+
+#define L2LMCAP		0x71D00
+#define L2MPAT		0x71D04
+
+/* These registers are for the L2 HP SRAM bank power management control and status.*/
+#define L2HSBPM_REG		L2LMCAP + 0x0010
```
Comment (softwarecki) [inline soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:90]: Maybe it's worth creating a structure for this group of registers?

File: soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h [hunk 2]
```diff
@@ -65,5 +65,39 @@
...
+struct cavs_lpsram_regs {
+	/** @brief power gating control */
+	uint32_t USxPGCTL;
+	/** @brief retention mode control */
+	uint32_t USxRMCTL;
+	/** @brief power gating status */
+	uint32_t USxPGISTS;
+};
+
+#define L2LMCAP		0x71D00
```
Comment (softwarecki) [inline soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:86]: Some comment what is it?

**Review by softwarecki — COMMENTED** [review]
Some of my nitpick in comments. Its a good practice to enclose all calculations of macro value in brackets.

**Review by kv2019i — DISMISSED** [review]
Makes sense to align the Intel targets.

**Comment by marc-hb** [comment]
Failures in https://github.com/zephyrproject-rtos/zephyr/actions/runs/8297862086/job/22710565918?pr=70221
seem relevant.

```
INFO    - 260/721 intel_adsp/cavs25/tgph    samples/synchronization/sample.kernel.synchronization  ERROR Build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/intel_adsp_cavs25_tgph/samples/synchronization/sample.kernel.synchronization/build.log
ERROR   - Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/samples/synchronization


FAILED: zephyr/CMakeFiles/zephyr.dir/soc/intel/intel_adsp/cavs/power_down_cavs.S.obj 


/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h: Assembler messages:
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:68: Error: unknown opcode or format name 'struct'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:70: Error: unknown opcode or format name 'uint32_t'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:72: Error: unknown opcode or format name 'uint32_t'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:74: Error: unknown opcode or format name 'uint32_t'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:75: Error: extra closing brace
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:75: Error: junk at end of line, first unrecognized character is `}'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:77: Error: unknown opcode or format name 'struct'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:79: Error: unknown opcode or format name 'uint32_t'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:81: Error: unknown opcode or format name 'uint32_t'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:83: Error: unknown opcode or format name 'uint32_t'
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:84: Error: extra closing brace
/__w/zephyr/zephyr/soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:84: Error: junk at end of line, first unrecognized character is `}'
ninja: build stopped: subcommand failed.

```

File: soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h [hunk 3]
```diff
@@ -65,5 +65,43 @@
...
+	/** @brief power gating status */
+	uint32_t USxPGISTS;
+};
+#endif /* _ASMLANGUAGE */
+
+#define L2LMCAP		0x71D00
+#define L2MPAT		0x71D04
+
+/* These registers are for the L2 HP SRAM bank power management control and status.*/
+#define L2HSBPM_REG		(L2LMCAP + 0x0010)
```
Comment (andyross) [inline soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h:94]: I know things are a little mixed with this hardware, but register block addresses really want to be in devicetree and not headers.

**Comment by marc-hb** [comment]
The ` #include "object.inc"` failure https://github.com/zephyrproject-rtos/zephyr/pull/70221/checks?check_run_id=22825300350 is unrelated and has just been fixed.

To restart CI, update the commit date and force-push (e.g.: `git commit -C HEAD --amend`)

No need to rebase contrary to a very common misconception. CI always merges with the moving target branch (that's how it detects conflicts).

Rebase when there are git conflicts (none now) and avoid rebase noise:
https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#workflow-suggestions-that-help-reviewers

File: soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h (hunk 3 above)
Comment (dnikodem) [reply soc/intel/intel_adsp/cavs/include/cavs25/adsp_memory.h]: Fixed

**Review by tmleman — DISMISSED** [review]

**Review by kv2019i — DISMISSED** [review]

**Comment by dnikodem** [comment]
@kv2019i  Please, do not merge this yet - found some gap in dtsi file

**Review by kv2019i — APPROVED** [review]

**Review by pjdobrowolski — APPROVED** [review]

**Review by iganakov — APPROVED** [review]

**Review by dabekjakub — APPROVED** [review]

**Comment by dnikodem** [comment]
@nashif This would be ready to go, thanks.

**Review by tmleman — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**